common.go

```diff
@@ -47,10 +54,15 @@ type AdditionalDetails map[string]string
 
 // ThreeDSecureAttributes is a set of attributes for 3D-Secure.
 type ThreeDSecureAttributes struct {
-	Encoding string `json:"encoding"`
-	XID      string `json:"xid"`
-	CAVV     string `json:"cavv"`
-	EciFlag  string `json:"eci_flag"`
+	Internal ThreeDSecureAttributesInternal `json:"internal"`
+}
+
+type ThreeDSecureAttributesInternal struct {
+	Encoding                   string                            `json:"encoding,omitempty"`
+	XID                        string                            `json:"xid,omitempty"`
+	CAVV                       string                            `json:"cavv,omitempty"`
```

Comment:
I don't see these attributes in "internal" section of "three_d_secure_attrributes". https://developers.paymentsos.com/docs/apis/payments/1.3.0/#operation/create-an-authorization.
Looks like you took them from "external" by mistake.

Reply:
this struct is used for several requests, in some cases in contains different fields of this struct. I use omit empty to allow to use only needed